brokers/vscode/broker.go

```diff
@@ -0,0 +1,296 @@
+//
+// Copyright (c) 2012-2018 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package vscode
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/eclipse/che-go-jsonrpc"
+	"github.com/eclipse/che-plugin-broker/common"
+	"github.com/eclipse/che-plugin-broker/files"
+	"github.com/eclipse/che-plugin-broker/model"
+	"github.com/eclipse/che-plugin-broker/storage"
+)
+
+const prefix = "vscode:extension/"
+const marketplace = "https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery"
+const bodyFmt = `{"filters":[{"criteria":[{"filterType":7,"value":"%s"}],"pageNumber":1,"pageSize":1,"sortBy":0, "sortOrder":0 }],"assetTypes":["Microsoft.VisualStudio.Services.VSIXPackage"],"flags":131}`
+const assetType = "Microsoft.VisualStudio.Services.VSIXPackage"
+
+// VSCodeExtensionBroker is used to process VS Code extensions to run them as Che plugins
+type VSCodeExtensionBroker struct {
+	common.Broker
+	ioUtil  files.IoUtil
+	Storage *storage.Storage
+}
+
+// NewBroker creates Che VS Code extension broker instance
+func NewBroker() *VSCodeExtensionBroker {
+	return &VSCodeExtensionBroker{
+		common.NewBroker(),
+		files.New(),
+		storage.New(),
+	}
+}
+
+// Start executes plugins metas processing and sends data to Che master
+func (broker *VSCodeExtensionBroker) Start(metas []model.PluginMeta) {
+	if ok, status := broker.Storage.SetStatus(model.StatusStarted); !ok {
+		m := fmt.Sprintf("Starting broker in state '%s' is not allowed", status)
+		broker.PubFailed(m)
+		broker.PrintFatal(m)
+	}
+	broker.PubStarted()
+	broker.PrintInfo("Started Plugin Broker")
```

Comment:
I noted that all plugin brokers publish this line and it is confusing when several plugin brokers are run at the same time:
![screenshot_20181228_134329](https://user-images.githubusercontent.com/5887312/50514619-fa7fb380-0aa7-11e9-8530-ba1dfb337268.png)
Please consider changing it to `Started VSCode Plugin Broker` here and the corresponding changes in other plugin brokers

Reply:
ok